test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -184,11 +205,22 @@ public void stop() {
 
     public void start(final boolean wait, final ServiceSpec serviceSpec) {
         try {
-            ServiceCreateResponse serviceCreateResponse = Exceptions.handleInterruptedCall(() -> dockerClient.createService(serviceSpec));
+            // List the service with filter 'serviceName'
+            Service.Criteria servicecriteria = Service.Criteria.builder().serviceName(serviceName).build();
+            List<Service> serviceList = Exceptions.handleInterruptedCall(
+                    () -> dockerClient.listServices(servicecriteria));
+            log.info("Service list size {}", serviceList.size());
+            if (!serviceList.isEmpty()) {
+                 String serviceId = serviceList.get(0).id();
```

Comment:
Can we avoid invoking dockerClient.listServices() to fetch the id ? 
If the id is not a constant then this code snippet of fetching id can be refactored out to a separate method.

Reply:
serviceId is not constant. Added separate method for fetching id and replicas.